CHANGELOG.md

```diff
@@ -52,6 +52,7 @@ RPM packages, which has been left unchanged.
 1. [20634](https://github.com/influxdata/influxdb/pull/20634): Promote schema and fill query optimizations to default behavior.
 1. [20678](https://github.com/influxdata/influxdb/pull/20678): Upgrade Flux to v0.104.0
 1. [20680](https://github.com/influxdata/influxdb/pull/20680): UI: Upgrade flux-lsp-browser to v0.5.31
+1. [20307](https://github.com/influxdata/influxdb/pull/20307): Add `retry-failed` cmd to CLI to rerun failed runs. Also, refactored [task.go](https://github.com/influxdata/influxdb/blob/master/cmd/influx/task.go) to follow builder pattern for testing.
```

Comment:
```suggestion
1. [20307](https://github.com/influxdata/influxdb/pull/20307): Add `influx task retry-failed` command to rerun failed runs.
```

Reply:
This line also needs to move up to the "unreleased" section now that 2.0.4 has been cut

Reply:
Some explanation @Nav-aggarwal09: A change log like ours is usually intended for users fo the product, not developers, so a change to internals and unit tests is not relevant. In general, use the change log to inform end users of a change with external impact like a new feature, and the commit log to inform other developers about internal changes like test refactoring.  

(Correct me if I'm wrong about your reasoning @danxmoran)